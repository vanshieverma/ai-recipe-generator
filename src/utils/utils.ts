import axios from "axios";
import { getSession } from 'next-auth/react';
import { ExtendedRecipe } from "../types";
import { GetServerSidePropsContext } from "next";

// Filters the results by enhancing recipe information with ownership and liked status for the user
export const filterResults = (recipes: ExtendedRecipe[], userId: string) => {
  return recipes.map((recipe) => (
    {
      ...recipe,
      owner: {
        _id: recipe.owner._id,
        name: recipe.owner.name,
        image: recipe.owner.image
      },
      likedBy: recipe.likedBy.map(({ _id, name, image }) => ({ _id, name, image })), // Simplifies likedBy list
      owns: recipe.owner._id.toString() === userId, // Flags if the recipe belongs to the user
      liked: recipe.likedBy.some(l => l._id.toString() === userId) // Flags if the user liked the recipe
    }
  ))
}

// Updates the recipe list by either replacing or removing a recipe from the list
export const updateRecipeList = (oldList: ExtendedRecipe[], newRecipe: ExtendedRecipe | null, deleteId?: string) => {
  const indexOfUpdate = oldList.findIndex((p) => p._id === (newRecipe ? newRecipe._id : deleteId));
  return newRecipe ? [
    ...oldList.slice(0, indexOfUpdate), // Preserves recipes before the updated one
    newRecipe, // Inserts the updated recipe
    ...oldList.slice(indexOfUpdate + 1), // Preserves recipes after the updated one
  ] : [
    ...oldList.slice(0, indexOfUpdate), // Preserves recipes before the deleted one
    ...oldList.slice(indexOfUpdate + 1), // Removes the deleted recipe
  ];
};

// Filters recipes based on search criteria in name, ingredients, or dietary preferences
export const getFilteredRecipes = (recipes: ExtendedRecipe[], search: string | null) => {
  if (!search) return recipes;
  const filteredRecipes = recipes.filter(({ name, ingredients, dietaryPreference }) => {
    const isFoundInName = name.toLowerCase().includes(search); // Matches search with recipe name
    const isFoundInIngredients = ingredients.filter(ingredient => ingredient.name.toLowerCase().includes(search)); // Matches search with ingredients
    const isFoundInDiets = dietaryPreference.filter(diet => diet.toLowerCase().includes(search)); // Matches search with dietary preferences
    return isFoundInName || Boolean(isFoundInIngredients.length) || Boolean(isFoundInDiets.length);
  });
  return filteredRecipes;
};

// Utility to fetch data on server-side while ensuring user authentication
export const getServerSidePropsUtility = async (context: GetServerSidePropsContext, address: string, propskey: string = 'recipes') => {
  try {
    const session = await getSession(context);
    if (!session) {
      return {
        redirect: {
          destination: '/',
          permanent: false,
        },
      };
    }
    const { data } = await axios.get(`${process.env.NEXT_PUBLIC_API_BASE_URL}/${address}`, {
      headers: {
        Cookie: context.req.headers.cookie || '',
      },
    });
    return {
      props: {
        [propskey]: data,
      },
    };
  } catch (error) {
    console.error(`Failed to fetch ${propskey}:`, error); // Logs errors in fetching data
    return {
      props: {
        [propskey]: [], // Returns an empty list if there's an error
      },
    };
  }
};

// REST API call utility supporting multiple HTTP methods
interface methods {
  put: 'put';
  post: 'post';
  delete: 'delete';
  get: 'get';
}

interface RESTcallTypes {
  address: string;
  method?: keyof methods;
  payload?: {
    [key: string]: any;
  };
}

export const call_api = async ({ address, method = 'get', payload }: RESTcallTypes) => {
  try {
    const { data } = await axios[method as keyof methods](address, payload);
    return data; // Returns the data from the API call
  } catch (error) {
    console.error(`An error occurred making a ${method} REST call to -> ${address} error -> ${error}`);
    throw (error); // Rethrows the error for further handling
  }
};
