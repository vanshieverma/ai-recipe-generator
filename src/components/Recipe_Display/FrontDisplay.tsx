import Image from "next/image"
import { Button } from '@headlessui/react'
import { call_api } from "../../utils/utils";
import { ExtendedRecipe } from '../../types';
import Loading from "../Loading";


interface FrontDisplayProps {
    recipe: ExtendedRecipe
    showRecipe: (recipe: ExtendedRecipe) => void
    updateRecipeList: (recipe: ExtendedRecipe) => void
    isLoading: boolean
}

const getThumbsup = ({ liked, owns }: { liked: boolean, owns: boolean }) => {
    if (owns) {
        return (<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="gray" className="size-5">
            <path d="M7.493 18.5c-.425 0-.82-.236-.975-.632A7.48 7.48 0 0 1 6 15.125c0-1.75.599-3.358 1.602-4.634.151-.192.373-.309.6-.397.473-.183.89-.514 1.212-.924a9.042 9.042 0 0 1 2.861-2.4c.723-.384 1.35-.956 1.653-1.715a4.498 4.498 0 0 0 .322-1.672V2.75A.75.75 0 0 1 15 2a2.25 2.25 0 0 1 2.25 2.25c0 1.152-.26 2.243-.723 3.218-.266.558.107 1.282.725 1.282h3.126c1.026 0 1.945.694 2.054 1.715.045.422.068.85.068 1.285a11.95 11.95 0 0 1-2.649 7.521c-.388.482-.987.729-1.605.729H14.23c-.483 0-.964-.078-1.423-.23l-3.114-1.04a4.501 4.501 0 0 0-1.423-.23h-.777ZM2.331 10.727a11.969 11.969 0 0 0-.831 4.398 12 12 0 0 0 .52 3.507C2.28 19.482 3.105 20 3.994 20H4.9c.445 0 .72-.498.523-.898a8.963 8.963 0 0 1-.924-3.977c0-1.708.476-3.305 1.302-4.666.245-.403-.028-.959-.5-.959H4.25c-.832 0-1.612.453-1.918 1.227Z" />
        </svg>)
    }
    if (liked) {
        return (<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="#0070ff" className="size-5">
            <path d="M7.493 18.5c-.425 0-.82-.236-.975-.632A7.48 7.48 0 0 1 6 15.125c0-1.75.599-3.358 1.602-4.634.151-.192.373-.309.6-.397.473-.183.89-.514 1.212-.924a9.042 9.042 0 0 1 2.861-2.4c.723-.384 1.35-.956 1.653-1.715a4.498 4.498 0 0 0 .322-1.672V2.75A.75.75 0 0 1 15 2a2.25 2.25 0 0 1 2.25 2.25c0 1.152-.26 2.243-.723 3.218-.266.558.107 1.282.725 1.282h3.126c1.026 0 1.945.694 2.054 1.715.045.422.068.85.068 1.285a11.95 11.95 0 0 1-2.649 7.521c-.388.482-.987.729-1.605.729H14.23c-.483 0-.964-.078-1.423-.23l-3.114-1.04a4.501 4.501 0 0 0-1.423-.23h-.777ZM2.331 10.727a11.969 11.969 0 0 0-.831 4.398 12 12 0 0 0 .52 3.507C2.28 19.482 3.105 20 3.994 20H4.9c.445 0 .72-.498.523-.898a8.963 8.963 0 0 1-.924-3.977c0-1.708.476-3.305 1.302-4.666.245-.403-.028-.959-.5-.959H4.25c-.832 0-1.612.453-1.918 1.227Z" />
        </svg>)
    }
    return (
        <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" strokeWidth={1.5} stroke="#0070ff" className="size-5">
            <path strokeLinecap="round" strokeLinejoin="round" d="M6.633 10.25c.806 0 1.533-.446 2.031-1.08a9.041 9.041 0 0 1 2.861-2.4c.723-.384 1.35-.956 1.653-1.715a4.498 4.498 0 0 0 .322-1.672V2.75a.75.75 0 0 1 .75-.75 2.25 2.25 0 0 1 2.25 2.25c0 1.152-.26 2.243-.723 3.218-.266.558.107 1.282.725 1.282m0 0h3.126c1.026 0 1.945.694 2.054 1.715.045.422.068.85.068 1.285a11.95 11.95 0 0 1-2.649 7.521c-.388.482-.987.729-1.605.729H13.48c-.483 0-.964-.078-1.423-.23l-3.114-1.04a4.501 4.501 0 0 0-1.423-.23H5.904m10.598-9.75H14.25M5.904 18.5c.083.205.173.405.27.602.197.4-.078.898-.523.898h-.908c-.889 0-1.713-.518-1.972-1.368a12 12 0 0 1-.521-3.507c0-1.553.295-3.036.831-4.398C3.387 9.953 4.167 9.5 5 9.5h1.053c.472 0 .745.556.5.96a8.958 8.958 0 0 0-1.302 4.665c0 1.194.232 2.333.654 3.375Z" />
        </svg>
    )
}


function FrontDisplay({ recipe, showRecipe, updateRecipeList, isLoading }: FrontDisplayProps) {

    const handleRecipeLike = async (recipeId: string) => {
        try {
            const result = await call_api({ address: '/api/like-recipe', method: 'put', payload: { recipeId } })
            updateRecipeList(result);
        } catch (error) {
            console.log(error)
        }
    }

    return (
        <div className="max-w-sm bg-gradient-to-r from-slate-200 to-stone-100 border border-gray-200 rounded-lg shadow-lg mt-4 mb-2 transform transition-transform hover:scale-105 hover:shadow-lg flex flex-col h-full">
            <div className="relative w-full h-64"> {/* Add a container for the image */}
                <Image
                    src={recipe.imgLink}
                    fill
                    alt={recipe.name}
                    style={{ objectFit: 'cover' }}
                    className="rounded-t-lg"
                    priority
                    sizes="auto"
                />
            </div>
            <div className="p-5 flex-grow">
                <h5 className="mb-2 text-2xl font-bold tracking-tight text-gray-900 drop-shadow-lg">{recipe.name}</h5>
                <p className="font-normal text-gray-700 dark:text-gray-400">{recipe.additionalInformation.nutritionalInformation}</p>
            </div>
            {
                isLoading ?
                    <div className="p-10">
                        <Loading />
                    </div>
                    :
                    <>
                        <div className="mx-auto flex">
                            {
                                recipe.dietaryPreference.map((preference) => (
                                    <span key={preference} className="chip bg-green-100 text-green-800 text-sm font-medium me-2 px-2.5 py-0.5 rounded hover:scale-110">{preference}</span>
                                ))
                            }
                        </div>
                        <div className="p-5">
                            <div className="flex items-center justify-between">
                                <Button
                                    className="inline-flex items-center px-3 py-2 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
                                    onClick={() => showRecipe(recipe)}
                                >
                                    See Recipe
                                    <svg className="rtl:rotate-180 w-3.5 h-3.5 ms-2" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 14 10">
                                        <path stroke="currentColor" strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M1 5h12m0 0L9 1m4 4L9 9" />
                                    </svg>
                                </Button>
                                <Button
                                    className="py-1.5 px-3 hover:text-blue-600 hover:scale-105 hover:shadow text-center border border-gray-300 rounded-md border-gray-400 h-8 text-sm flex items-center gap-1 lg:gap-2"
                                    onClick={() => handleRecipeLike(recipe._id)}
                                    disabled={recipe.owns}
                                    data-testid="like_button"
                                >
                                    {getThumbsup(recipe)}
                                    <span>{recipe.likedBy.length}</span>
                                </Button>
                            </div>
                        </div>
                    </>
            }

        </div>

    )
}

export default FrontDisplay


