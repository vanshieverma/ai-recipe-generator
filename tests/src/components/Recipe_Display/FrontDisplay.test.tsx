import FrontDisplay from "../../../../src/components/Recipe_Display/FrontDisplay";
import { render, screen, fireEvent } from '@testing-library/react'
import * as apiCalls from "../../../../src/utils/utils";
import { stub_recipe_1 } from "../../../stub";

jest.mock("../../../../src/utils/utils")

describe('The single front facing display', () => {
    let updatedRecipe: any;
    const updateRecipeListMock = jest.fn();
    beforeEach(() => {
        updatedRecipe = {
            ...stub_recipe_1,
            owns: false,
            liked: false
        }
    })
    it('shall render', () => {
        const { container } = render(<FrontDisplay recipe={updatedRecipe} showRecipe={jest.fn()} updateRecipeList={jest.fn()} isLoading={false} />)
        expect(container).toMatchSnapshot()
    })

    it('shall render for a loading recipe', () => {
        const { container } = render(<FrontDisplay recipe={updatedRecipe} showRecipe={jest.fn()} updateRecipeList={jest.fn()} isLoading={true} />)
        expect(container).toMatchSnapshot()
    })

    it('shall handle "Liking" a recipe', async () => {
        const likeRecipe = jest.spyOn(apiCalls, 'call_api');
        likeRecipe.mockImplementationOnce(() => Promise.resolve({
            ...updatedRecipe,
            liked: true
        }))
        const { container } = render(<FrontDisplay recipe={updatedRecipe} showRecipe={jest.fn()} updateRecipeList={updateRecipeListMock} isLoading={false} />)
        fireEvent.click(screen.getByTestId('like_button'));
        await screen.findByText('Recipe_1_name')
        expect(updateRecipeListMock).toHaveBeenCalledWith({ ...updatedRecipe, liked: true })
        expect(likeRecipe).toHaveBeenCalledWith({
            "address": "/api/like-recipe",
            "method": "put",
            "payload": { "recipeId": "6683b8d38475eac9af5fe838" }
        })
    })

})