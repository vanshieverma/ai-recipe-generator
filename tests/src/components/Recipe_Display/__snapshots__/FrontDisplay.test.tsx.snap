// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The single front facing display shall render 1`] = `
<div>
  <div
    class="max-w-sm bg-gradient-to-r from-slate-200 to-stone-100 border border-gray-200 rounded-lg shadow-lg mt-4 mb-2 transform transition-transform hover:scale-105 hover:shadow-lg flex flex-col h-full"
  >
    <div
      class="relative w-full h-64"
    >
       
      <img
        alt="Recipe_1_name"
        class="rounded-t-lg"
        data-nimg="fill"
        decoding="async"
        fetchpriority="high"
        sizes="auto"
        src="/_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=16&q=75 16w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=32&q=75 32w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=48&q=75 48w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=64&q=75 64w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=96&q=75 96w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=128&q=75 128w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
      />
    </div>
    <div
      class="p-5 flex-grow"
    >
      <h5
        class="mb-2 text-2xl font-bold tracking-tight text-gray-900 drop-shadow-lg"
      >
        Recipe_1_name
      </h5>
      <p
        class="font-normal text-gray-700 dark:text-gray-400"
      >
        Recipe_1_nutritionalInformation
      </p>
    </div>
    <div
      class="mx-auto flex"
    >
      <span
        class="chip bg-green-100 text-green-800 text-sm font-medium me-2 px-2.5 py-0.5 rounded hover:scale-110"
      >
        Recipe_1_preference_1
      </span>
      <span
        class="chip bg-green-100 text-green-800 text-sm font-medium me-2 px-2.5 py-0.5 rounded hover:scale-110"
      >
        Recipe_1_preference_2
      </span>
    </div>
    <div
      class="p-5"
    >
      <div
        class="flex items-center justify-between"
      >
        <button
          class="inline-flex items-center px-3 py-2 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
          data-headlessui-state=""
          type="button"
        >
          See Recipe
          <svg
            aria-hidden="true"
            class="rtl:rotate-180 w-3.5 h-3.5 ms-2"
            fill="none"
            viewBox="0 0 14 10"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 5h12m0 0L9 1m4 4L9 9"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
        <button
          class="py-1.5 px-3 hover:text-blue-600 hover:scale-105 hover:shadow text-center border border-gray-300 rounded-md border-gray-400 h-8 text-sm flex items-center gap-1 lg:gap-2"
          data-headlessui-state=""
          data-testid="like_button"
          type="button"
        >
          <svg
            class="size-5"
            fill="none"
            stroke="#0070ff"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.633 10.25c.806 0 1.533-.446 2.031-1.08a9.041 9.041 0 0 1 2.861-2.4c.723-.384 1.35-.956 1.653-1.715a4.498 4.498 0 0 0 .322-1.672V2.75a.75.75 0 0 1 .75-.75 2.25 2.25 0 0 1 2.25 2.25c0 1.152-.26 2.243-.723 3.218-.266.558.107 1.282.725 1.282m0 0h3.126c1.026 0 1.945.694 2.054 1.715.045.422.068.85.068 1.285a11.95 11.95 0 0 1-2.649 7.521c-.388.482-.987.729-1.605.729H13.48c-.483 0-.964-.078-1.423-.23l-3.114-1.04a4.501 4.501 0 0 0-1.423-.23H5.904m10.598-9.75H14.25M5.904 18.5c.083.205.173.405.27.602.197.4-.078.898-.523.898h-.908c-.889 0-1.713-.518-1.972-1.368a12 12 0 0 1-.521-3.507c0-1.553.295-3.036.831-4.398C3.387 9.953 4.167 9.5 5 9.5h1.053c.472 0 .745.556.5.96a8.958 8.958 0 0 0-1.302 4.665c0 1.194.232 2.333.654 3.375Z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <span>
            1
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`The single front facing display shall render for a loading recipe 1`] = `
<div>
  <div
    class="max-w-sm bg-gradient-to-r from-slate-200 to-stone-100 border border-gray-200 rounded-lg shadow-lg mt-4 mb-2 transform transition-transform hover:scale-105 hover:shadow-lg flex flex-col h-full"
  >
    <div
      class="relative w-full h-64"
    >
       
      <img
        alt="Recipe_1_name"
        class="rounded-t-lg"
        data-nimg="fill"
        decoding="async"
        fetchpriority="high"
        sizes="auto"
        src="/_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=16&q=75 16w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=32&q=75 32w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=48&q=75 48w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=64&q=75 64w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=96&q=75 96w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=128&q=75 128w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fsmart-recipe-generator.s3.amazonaws.com%2FRecipe_1&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
      />
    </div>
    <div
      class="p-5 flex-grow"
    >
      <h5
        class="mb-2 text-2xl font-bold tracking-tight text-gray-900 drop-shadow-lg"
      >
        Recipe_1_name
      </h5>
      <p
        class="font-normal text-gray-700 dark:text-gray-400"
      >
        Recipe_1_nutritionalInformation
      </p>
    </div>
    <div
      class="p-10"
    >
      <div
        class="flex items-center justify-center mt-5"
      >
        <div
          class="relative"
        >
          <div
            class="h-24 w-24 rounded-full border-t-8 border-b-8 border-gray-200"
          />
          <div
            class="absolute top-0 left-0 h-24 w-24 rounded-full border-t-8 border-b-8 border-blue-500 animate-spin"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
